Engine/Helper.cs

```diff
@@ -656,22 +656,46 @@ public IScriptExtent GetScriptExtentForFunctionName(FunctionDefinitionAst functi
             if (null == functionDefinitionAst)
             {
                 return null;
+            }            
+            var funcNameToken = Tokens.Where(
+                token => ContainsExtent(functionDefinitionAst.Extent, token.Extent) 
+                && token.Text.Equals(functionDefinitionAst.Name));
+            if (funcNameToken.Any())
```

Comment:
We return the extent property of the first element and not the first element itself. Assuming all the tokens are non-null, If we just use FirstOrDefault we still need to check for null (if no token was found) before getting the Extent property.